### ISSUE TITLE
Listblock offset allows negative values that break the page

### ISSUE THREAD

**Issue by e-marie-w** [uiowa/uiowa#6899]
# Problem
Page displays generic 'There was a problem' error when clicking view or layout tabs for a page with a negative offset listblock. Locally, error specifies problem in the database with sql syntax.

This came into the support inbox, I was able to fix it for that site by deleting the node tempstore and any revisions with the negative value.

# Steps to reproduce
Add or edit a listblock - I've only tested Articles and People - and give it a negative offset. Add/save block, then close configure block tray (since it won't on its own). Save layout. You will now see the error. Go to /admin/content. Click edit for the page with your listblock. Try the view and layout tabs, they will give the error. 

# Proposed solution
Limit offset values to anything equal or greater than zero.

**Comment by joewhitsitt** [comment]
Gosh, I thought we dealt with this previously.. but I think it was just the "items_per_page" setting.

https://github.com/uiowa/uiowa/blob/main/docroot/modules/custom/layout_builder_custom/src/Plugin/Display/ListBlock.php#L160-L168
```php
    // Modify "Items per page" block settings form.
    if (!empty($allow_settings['items_per_page'])) {
      // @todo Remove once exposed filters patch is added.
      // Seems to break at high numbers :grimmacing: ..
      $form['override']['items_per_page']['#min'] = 1;
      $form['override']['items_per_page']['#max'] = 50;
      $form['override']['items_per_page']['#title'] = $this->t('Items to display');
      $form['override']['items_per_page']['#description'] = $this->t('Select the number of entries to display. Minimum of 1 and maximum of 50. Show pager to display more than 50.');
    }
```

Looks like we fixed it for v2 https://github.com/uiowa/uiowa/pull/1969.

**Comment by joewhitsitt** [comment]
I can't seem to find something specific to offset, but it seems there are issues related to the pager:

https://www.drupal.org/project/drupal/issues/2871784
https://www.drupal.org/project/drupal/issues/3080859

**Comment by briand44** [comment]
Grooming conversation... Probably need an update hook to account for any we find that are currently set to a negative value. Update to set it to 0.

**Comment by pyrello** [comment]
There was a fix that went out in #7087. Not sure if that fully resolve this or not.
